packages/core/src/browser/context-menu-service.ts

```diff
@@ -0,0 +1,56 @@
+/********************************************************************************
+ * Copyright (C) 2019 TypeFox and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { TabBar, Widget, Title } from '@phosphor/widgets';
+import { injectable, inject } from 'inversify';
+import { ApplicationShell } from './shell/application-shell';
+
+@injectable()
+export class ContextMenuService {
```

Comment:
It looks weird. `ContextMenuService` will be responsible to open and close context menus, not a bunch of utility methods.

That should belong to `ApplicationShell`.

Reply:
I think we should just delete this class and rewrite commands with existing APIs, there are enough already of them.